.github/workflows/generated_cmake.yml

```diff
@@ -12,13 +17,17 @@ jobs:
       fail-fast: false   # Don't cancel all jobs if one fails.
 
     steps:
-      - uses: actions/checkout@v2
-      - name: Set up Bazel read-only caching
-        run: echo "BAZEL_CACHE_AUTH=--remote_upload_local_results=false" >> $GITHUB_ENV
-      - name: Generate CMake files
-        run: cd ${{ github.workspace }} && bazel build //pkg:gen_src_file_lists --test_output=errors $BAZEL_CACHE $BAZEL_CACHE_AUTH
-      - name: Compare to Golden file
-        run: diff -du bazel-bin/pkg/src_file_lists.cmake src/file_lists.cmake
-      - name: Report
-        run: echo "::error file=cmake/update_file_lists.sh::CMake files are stale, please run cmake/update_file_lists.sh"
-        if: failure()
+      - uses: actions/checkout@v3
+        with:
+          token: ${{ secrets.BOT_ACCESS_TOKEN }}
+      - name: Configure name and email address in Git
+        run: cd ${{ github.workspace }} && git config user.name "Protobuf Team Bot" && git config user.email "protobuf-team-bot@google.com"
+      - name: Generate CMake file lists
+        run: cd ${{ github.workspace }} && ./cmake/update_file_lists.sh
+      - name: Guard against infinite loop
+        # As an extra precaution, we undo any pending changes if the most
+        # recent commit was made by the bot. This reduces the risk of a bug
+        # causing an infinite loop of commits.
+        run: if (git log -1 --pretty=format:'%an' | grep -q "Protobuf Team Bot"); then git reset --hard HEAD; fi
+      - name: Commit and push change
+        run: cd ${{ github.workspace }} && git add -A && (git diff --staged --quiet || git commit -am "Auto-generate CMake file lists") && git push
```

Comment:
I feel like this could be reworded to make it clearer what the commit did and why.  e.g.
- "Auto-refresh CMake configuration after PR #{pr number}"
- "Auto-generate new CMake configution after PR #{pr number}"

Reply:
I'm leaning towards the second one just so we can unify all our bot-generated commits with the prefix "auto-generated"

Reply:
Ok, let me give that a try. I think this logic is getting complicated enough that I should probably move it into a shell script at this point, and that should make it much easier to do than if I did it all directly in the yml file.

Reply:
sgtm, I think you need to update this to run it?

Reply:
Yes, this is now ready to go and has the more detailed commit message.